src/components/questionnaire/questionnaire-table.tsx

```diff
@@ -106,6 +107,10 @@ const QuestionnaireTable = () => {
    * @param params
    */
   const handleRowClick = (params: GridRowParams) => {
+    const userHasPassed = params.row.passedUsers?.includes(loggedInUser?.id);
+    if (userHasPassed) {
+      return;
+    }
```

Comment:
Interesting didn't think about this.
I think a user should be able to complete a questionnaire again even if they have passed it. For now I'd just add a TODO comment here to it can be resolved in another feature.